gabarit/template_nlp/nlp_project/tests/test_model_huggingface.py

```diff
@@ -874,6 +950,74 @@ def test16_model_huggingface_load_standalone_files(self):
         remove_dir(model_dir)
         remove_dir(new_model.model_dir)
 
+    def test17_model_huggingface_fit_with_seed(self):
+        '''Test random seed for {{package_name}}.models_training.models_tensorflow.model_huggingface.ModelHuggingFace'''
+
+        model_dir = os.path.join(os.getcwd(), 'model_test_123456789')
+        remove_dir(model_dir)
+        model_dir2 = os.path.join(os.getcwd(), 'model_test_123456789_2')
+        remove_dir(model_dir2)
+
+        # Set vars
+        x_train = np.array(["ceci est un test", "pas cela", "cela non plus", "ici test", "là, rien!"] * 100)
+        y_train_mono = np.array([0, 1, 0, 1, 2] * 100)
+        y_train_multi = pd.DataFrame({'test1': [0, 0, 0, 1, 0] * 100, 'test2': [1, 0, 0, 0, 0] * 100, 'test3': [0, 0, 0, 1, 0] * 100})
+
+        # Mono-label same random_seed
+        model1 = ModelHuggingFace(model_dir=model_dir, batch_size=32, epochs=2, multi_label=False,
+                                  max_sequence_length=10, max_words=100, random_seed=42,
+                                  padding='pre', truncating='post',
+                                  embedding_name='fake_embedding.pkl')
+        model1.fit(x_train, y_train_mono)
+        model2 = ModelHuggingFace(model_dir=model_dir2, batch_size=32, epochs=2, multi_label=False,
+                                  max_sequence_length=10, max_words=100, random_seed=42,
+                                  padding='pre', truncating='post',
+                                  embedding_name='fake_embedding.pkl')
+        model2.fit(x_train, y_train_mono)
+        self.assertTrue(compare_transformers_models(model1.model, model2.model))
+        remove_dir(model_dir), remove_dir(model_dir2)
+
+        # Multi-label same random_seed
+        model1 = ModelHuggingFace(model_dir=model_dir, batch_size=32, epochs=2, multi_label=True,
+                                  max_sequence_length=10, max_words=100, random_seed=42,
+                                  padding='pre', truncating='post',
+                                  embedding_name='fake_embedding.pkl')
+        model1.fit(x_train, y_train_multi)
+        model2 = ModelHuggingFace(model_dir=model_dir2, batch_size=32, epochs=2, multi_label=True,
+                                  max_sequence_length=10, max_words=100, random_seed=42,
+                                  padding='pre', truncating='post',
+                                  embedding_name='fake_embedding.pkl')
+        model2.fit(x_train, y_train_multi)
+        self.assertTrue(compare_transformers_models(model1.model, model2.model))
+        remove_dir(model_dir), remove_dir(model_dir2)
+
+        # Mono-label different random_seed
```

Comment:
I would prefer the order: 
mono label same random seed
mono label different random seed
multi label same random seed
multi label different random seed

To mimick what was done in ModelKeras models